middlewares/access.middleware.js

```diff
@@ -0,0 +1,190 @@
+                res.locals.userData.access.vendor_payment_amount === undefined ||
+                res.locals.userData.access.customer_next_payment_date === undefined || 
+                res.locals.userData.access.customer_payment_amount === undefined ||
+                res.locals.userData.access.vendor_additional_paid === undefined
+            ) {
+                throw new Error("Missing Access Property");
+            }
+            var now = new Date();
+            switch (req.body.type.type) {
+                case 'VENDOR_SUBSCRIPTION':
+                    if (res.locals.userData.access.vendor_next_payment_date >= now) {
+                        errorResponse.send(res, 409, 'Vendor Access Already Active');
+                    } else {
+                        res.locals.vendor = true;
+                        next();
+                    }
+                    break;
+                case 'CUSTOMER_SUBSCRIPTION':
+                    if (res.locals.userData.access.customer_next_payment_date >= now) {
+                        errorResponse.send(res, 409, 'Customer Access Already Active');
+                    } else {
+                        res.locals.customer = true;
+                        next();
+                    }
+                    break;
+                case 'VENDOR_ADDITIONAL':
+                    if (res.locals.userData.access.vendor_next_payment_date < now) {
+                        errorResponse.send(res, 403, 'Vendor Access Required');
+                    } else if (res.locals.userData.access.vendor_additional_paid === true) {
+                        errorResponse.send(res, 409, 'Vendor Additional Already Paid');
+                    }else {
+                        res.locals.additional = true;
+                        next();
+                    }
+                    break;
+                default:
+                    errorResponse.send(res, 400, 'Malformed Request');
+                    break;
+            }
+        } catch (err) {
+            errorResponse.send(res, 500, 'Internal Server Error', err);
+        }
+    }
+
+    function createSubscription(req, res, next) {
+        if (res.locals.vendor === true || res.locals.customer === true) {
+            stripe.subscriptions.create({
+              customer: res.locals.userData.stripe_id,
+              items: [
+                {
+                  plan: req.body.type.type,
+                },
+              ],
+              source: req.body.token.id
+            }, function(err, subscription) {
+                if (err) {
+                    errorResponse.send(res, 500, 'Internal Server Error', err);
+                } else {
+                    if (res.locals.vendor === true) {
+                        res.locals.userData.access.vendor_payment_amount = subscription.plan.amount;
+                        res.locals.userData.access.vendor_next_payment_date = new Date(subscription.current_period_end * 1000);
```

Comment:
Should remove magic number 1000 to const

Reply:
it's just unit conversion from ms to s